docs/varnish/vcl/varnish4.vcl

```diff
@@ -167,6 +163,18 @@ sub ez_purge {
     }
 }
 
+sub ez_purge_acl {
+//    if (req.http.x-purge-token) {
+//        #  Won't work on Varnish <= 5.1, if needed in 4.1 you can hardcode a secret token here instead of std.getenv() usage
+//        if (req.http.x-purge-token != std.getenv("HTTPCACHE_VARNISH_INVALIDATE_TOKEN")) {
```

Comment:
I have not tested, but this looks like a security hole

If server is configured to use whitelist (invalidators), and evil user sends x-purge-token header with empty value, this `if` might match if env variable `HTTPCACHE_VARNISH_INVALIDATE_TOKEN` is not set?

Reply:
I think if `x-purge-token` will be empty - the if condition `if (req.http.x-purge-token) {` will return false and second condition (`req.http.x-purge-token != std.ge...`) will not be called

Reply:
This is why it's not enabled by default.
As enabling it means you should always inject env.

But, as p.sh does not support injecting it by env, which is where we need this. We can change example here to just have hardcoded token to avoid anyone stumbling into such an issue by mistake.

Avoid having to document any differences between 4.1 and higher as well.

Reply:
Or, what was the outcome so far of discussions with p.sh?

Reply:
No reply from ph.sh yet. I have created a ticket in their jira, requesting us to be able to define env variables in the varnish container

Reply:
I have now tested sending token header with empty value, and indeed it seems to work as we intended.

So, since that is the case, why comment out the lines in the first place...?

Reply:
See discussion above, it's because std.getenv is not defined on Varnish 4.1 (LTS).
We can drop support for 4.x in a future major, but not in a minor where we need this.

Reply:
acknowledged